pkg/piperenv/templating.go

```diff
@@ -0,0 +1,89 @@
+package piperenv
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"text/template"
+)
+
+// ParseTemplate allows to parse a template which contains references to the CPE
+// Utility functions make it simple to access specific parts of the CPE
+func (c *CPEMap) ParseTemplate(cpeTemplate string) (*bytes.Buffer, error) {
+	funcMap := template.FuncMap{
+		"cpe":         c.cpe,
+		"cpecustom":   c.custom,
+		"git":         c.git,
+		"imageDigest": c.imageDigest,
+		"imageTag":    c.imageTag,
+
+		// ToDo: add template function for artifacts
+		// This requires alignment on artifact handling before, though
+	}
+
+	tmpl, err := template.New("cpetemplate").Funcs(funcMap).Parse(cpeTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse cpe template '%v': %w", cpeTemplate, err)
+	}
+
+	tmplParams := struct {
+		CPE map[string]interface{}
+	}{
+		CPE: map[string]interface{}(*c),
+	}
+
+	var generated bytes.Buffer
+	err = tmpl.Execute(&generated, tmplParams)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute cpe template '%v': %w", cpeTemplate, err)
+	}
+
+	return &generated, nil
+}
+
+func (c *CPEMap) cpe(element string) string {
+	// ToDo: perform validity checks to allow only selected fields for now?
+	// This would allow a stable contract and could perform conversions in case a contract changes.
+
+	el, _ := map[string]interface{}(*c)[element].(string)
+	return el
+}
+
+func (c *CPEMap) custom(element string) string {
+	el, _ := map[string]interface{}(*c)[fmt.Sprintf("custom/%v", element)].(string)
```

Comment:
you are right. 
`imageTags` and `imageDigests` should not be a problem since we have explicit accessors to these fields.

What do you think, should we allow-list some fields where it makes sense to access?
We could then still add more in case we see demand?

Reply:
Yes, I think it makes sense.

Reply:
There was indeed a problem. I tried to fix it now with the latest commit.